resources/containers/field-modal.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
+import styles from "./styles/field-modal.module.css";
+
+interface FieldModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSave: (data: any) => void;
+  fieldType: string;
+}
+
+const FieldModal: React.FC<FieldModalProps> = ({
+  isOpen,
+  onClose,
+  onSave,
+  fieldType,
+}) => {
+  const [selectedOptions, setSelectedOptions] = useState<string[]>([]);
```

Comment:
Bad way to do this. Why we dont use react forms and yup? 
Then we have an hashmap type of property required inputs. 
At the end on save we just save the form result, no need any if-else